### ISSUE TITLE
IBus auxiliary processes not started

### ISSUE THREAD

**Issue by LRitzdorf** [KSXGitHub/ibus-daemon.pkgbuild#5]
When using `ibus@:0.service` to start IBus in on i3, the service succeeds in starting after login, but apparently does not spawn appropriate helper processes. In this state, IBus does not function for some programs, such as Firefox and even `ibus-setup`. However, restarting the service causes it to initialize properly.

The command output below should help illustrate this:
```
lucas at ritzcracker in ~ 
❯ systemctl --user status ibus@:0.service
● ibus@:0.service - Intelligent Input Bus
     Loaded: loaded (/usr/lib/systemd/user/ibus@.service; enabled; preset: enabled)
     Active: active (running) since Wed 2023-06-28 18:07:19 PDT; 21s ago
       Docs: man:ibus-daemon(1)
   Main PID: 1953 (ibus-daemon)
      Tasks: 10 (limit: 19053)
     Memory: 22.3M
        CPU: 101ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/app-ibus.slice/ibus@:0.service
             ├─1953 /usr/bin/ibus-daemon --replace --xim
             └─1992 /usr/lib/ibus/ibus-dconf

Jun 28 18:07:19 ritzcracker systemd[1945]: Started Intelligent Input Bus.

lucas at ritzcracker in ~ 
❯ systemctl --user restart ibus@:0.service

lucas at ritzcracker in ~ 
❯ systemctl --user status ibus@:0.service 
● ibus@:0.service - Intelligent Input Bus
     Loaded: loaded (/usr/lib/systemd/user/ibus@.service; enabled; preset: enabled)
     Active: active (running) since Wed 2023-06-28 18:10:10 PDT; 2s ago
       Docs: man:ibus-daemon(1)
   Main PID: 4211 (ibus-daemon)
      Tasks: 28 (limit: 19053)
     Memory: 38.8M
        CPU: 1.500s
     CGroup: /user.slice/user-1000.slice/user@1000.service/app.slice/app-ibus.slice/ibus@:0.service
             ├─4211 /usr/bin/ibus-daemon --replace --xim
             ├─4264 /usr/lib/ibus/ibus-dconf
             ├─4265 /usr/lib/ibus/ibus-ui-gtk3
             ├─4266 /usr/lib/ibus/ibus-extension-gtk3
             ├─4268 /usr/lib/ibus/ibus-x11 --kill-daemon
             └─4296 /usr/lib/ibus/ibus-engine-simple

Jun 28 18:10:10 ritzcracker systemd[1945]: Started Intelligent Input Bus.
```

From basic testing, I've found that adding the following service override fixes the issue for me:
```systemd
[Service]
ExecStartPre=/usr/bin/sleep 1
```

Does this point to an issue with my setup, or might it be worth adding to this package?

**Comment by KSXGitHub** [comment]
It works properly in Firefox on GNOME Xorg.

I don't use i3, so it's hard for me to reproduce this.

**Comment by LRitzdorf** [comment]
Fair enough. Is there any chance another maintainer might be able to test this, or should we close this issue as unique?

**Comment by KSXGitHub** [comment]
I am the only maintainer.

**Comment by shadowcomer** [comment]
I also ran into this issue.

The service shows as enabled, active, and without errors or warnings. However, the `i3wm` does not recognize it as running: the widget is not displayed, and the key combinations are not working.

I tried adding the delay to `ibus@.service`, but it did not solve the issue for me.

However, I did manage to workaround this issue by uninstalling the package, and instead starting `ibus` through my user's `.xinitrc` like the following:

```
# ... other configuration here
# At the end of the file:
ibus start -drxR
exec i3
```

Maybe `i3wm` isn't detecting `ibus` for some reason? Starting/restarting the service after having logged in seems to work, so it might not be entirely the service's fault. Maybe an initialization order issue?

I know this doesn't solve the actual issue, but I hope this helps anyone coming across this thread.

**Comment by KSXGitHub** [comment]
If more people can reproduce @shadowcomer's workaround, I will add the workaround to README.md.

**Comment by LRitzdorf** [comment]
(fwiw, launching ibus via xinitrc or your WM's startup configuration is the standard method, which this service was meant to be an alternative to. While totally valid, @shadowcomer's solution is actually just the normal way of doing things)

**Comment by shadowcomer** [comment]
> (fwiw, launching ibus via xinitrc or your WM's startup configuration is the standard method, which this service was meant to be an alternative to. While totally valid, @shadowcomer's solution is actually just the normal way of doing things)

Ah! I did not know this. Thanks for the heads up. I've found in the `ibus` repository [documentation that supports](https://github.com/ibus/ibus/wiki/IBusStartUpTroubleShooting) what you said, and explains a little bit more about how it works.

I originally ended up using the `ibus-daemon` package following [arch linux's `ibus` documentation](https://wiki.archlinux.org/title/IBus), which recommends it among other options for [autostarting](https://wiki.archlinux.org/title/Autostarting).

I guess that arch linux's documentation recommends the generic way of doing the setup (a user unit), which should work for users who have a desktop environment, but does not for those that only have a window manager. Maybe this should be reflected there, as it seems that this package works as is expected.
